orderer/common/multichannel/registrar.go

```diff
@@ -714,8 +741,10 @@ func (r *Registrar) JoinChannel(channelID string, configBlock *cb.Block, isAppCh
 
 	defer func() {
 		if err != nil {
-			if err2 := r.ledgerFactory.Remove(channelID); err2 != nil {
+			r.pendingRemoval[channelID] = true
+			if err2 := r.ledgerFactory.Remove(channelID, r.removeChannelFromPendingRemovals); err2 != nil {
```

Comment:
this could be done synchronously, as it is a small ledger, only just constructed